testing/todocheck_test.go

```diff
@@ -72,6 +72,26 @@ func TestFirstlineMalformedTodo(t *testing.T) {
 	}
 }
 
+func TestFirstlineMalformedTodoWithJsonOutput(t *testing.T) {
+	err := scenariobuilder.NewScenario().
+		WithBinary("../todocheck").
+		WithBasepath("./scenarios/firstline_comment").
+		WithConfig("./test_configs/no_issue_tracker.yaml").
+		WithJSONOutput().
+		ExpectTodoErr(
+			scenariobuilder.NewTodoErr().
+				WithType(errors.TODOErrTypeMalformed).
+				WithLocation("scenarios/firstline_comment/main.cpp", 1)).
+		ExpectTodoErr(
+			scenariobuilder.NewTodoErr().
+				WithType(errors.TODOErrTypeMalformed).
+				WithLocation("scenarios/firstline_comment/other.cpp", 1)).
```

Comment:
Could you also add a scenario for `TODOErrTypeNonExistent` and `TODOErrTypeClosed`.
This will guarantee that the JSON output is valid for those cases as well.

You can reuse the Jira localhost mock server, which is provided by the testing framework.
See the test `TestAnnotatedTodos` for an example.

Reply:
I added a new test function to test scenario for `TODOErrTypeNonExistent` and `TODOErrTypeClosed`.
https://github.com/preslavmihaylov/todocheck/pull/77/files#diff-4d213ea2bbf5c6daa9454def85352fba6ebdfe5860d3578dfc060a389bc15ce4R167